File: Python/MajorityElementTest.py
from unittest import TestCase

from MajorityElement import MajorityElement


class TestMajorityElement(TestCase):
    def test_majorityElement(self):
        me = MajorityElement()

        self.assertTrue(me.majorityElement([1, 1, 2]) == 1)

        self.assertTrue(me.majorityElement([0, 1, 1, 1, 2, 3, 1]) == 1)

        self.assertTrue(me.majorityElement([1, 1]) == 1)

File: Golang/MajorityElement/MajorityElement_test.go
package MajorityElement

import "testing"

func TestMajorityElement(t *testing.T) {

	if majorityElement([]int{1, 1, 2}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if majorityElement([]int{0, 1, 1, 1, 2, 3, 1}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if majorityElement([]int{1, 1}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}
